mergesort.rb

```diff
@@ -1,25 +1,45 @@
 def mergesort(a)
     # if the array size is 0|1 then the list is considered sorted, return sorted array
+    return a if a.length < 2
     # split the list into two halves
+    left,right = split_array(a)
     # merge sort each half
+    left = mergesort(left)
+    right = mergesort(right)
     # combine the sorted halves
+    combine(left,right)
 end
 
 def split_array(a)
+  split = a.each_slice((a.length/2.0).round).to_a
+  # probs don't need it but meh)
```

Comment:
It looks like this version of split_array works, but it's not immediately obvious how. In general, you don't get extra points for writing fewer lines of code. Try to write code that is clearer and more understandable to fellow programmers, even if it happens to be a few extra lines. Code will be written once and read many, many times. If a solution like this is the only way to do it, then some extra comments about how it does work are warranted.